server/src/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -2451,6 +2455,21 @@ public UserVm updateVirtualMachine(long id, String displayName, String group, Bo
             isDynamicallyScalable = vm.isDynamicallyScalable();
         }
 
+        // Get default guest network in Basic zone
+        DataCenterVO zone = _dcDao.findById(vm.getDataCenterId());
+        Network defaultNetwork = _networkModel.getExclusiveGuestNetwork(zone.getId());
+
+        boolean isVmWare = (vm.getHypervisorType() == HypervisorType.VMware);
```

Comment:
Small detail: Please check the spelling of VMware, I see lots of different variations and believe VMware is the right one.